### ISSUE TITLE
Yaml safe_load won't read source catalog schema.yaml file.

### ISSUE THREAD

**Issue by wmwv** [LSSTDESC/gcr-catalogs#302]
The `safe_load` of the current DC2 source catalog Run 1.2i schema.yaml fails:
```
import yaml
filepath = '/global/projecta/projectdirs/lsst/production/DC2_ImSim/Run1.2i/dpdd/source_catalog/schema.yaml'
with open(filepath, 'r') as schema_stream:
    foo = yaml.safe_load(schema_stream)
```
with
```
...
ConstructorError: could not determine a constructor for the tag 'tag:yaml.org,2002:python/object/apply:numpy.dtype'
  in "/global/projecta/projectdirs/lsst/production/DC2_ImSim/Run1.2i/dpdd/source_catalog/schema.yaml", line 2, column 10
```

I just pick this as a specific example.  The same problem is shared with the other schema files.

My fault, I should have tested this before I approved #299

**Comment by wmwv** [comment]
My current pessimistic reading is that we can't persist `numpy` objects in purely safe PyYaml load/dump.

**Comment by yymao** [comment]
Do we really need to have numpy object in the schema file?

Also, we should modify the reader so that it would still work if the schema reading fails.

**Comment by wmwv** [comment]
> Do we really need to have numpy object in the schema file?
Hmmm... that is a good question.  Why are those coming out?  It was generated by `generate_schema_yaml()` automatic creation.  Perhaps there's something like something being overly smart and recognizing that a column only has, e.g., 0, and 1 and defining that as a different type than float.

**Comment by yymao** [comment]
I should serialize dtype in generate_schema_yaml. Let me try to fix this in a PR.

**Comment by wmwv** [comment]
Ah, I see my speculation above wasn't right.  PyYaml is just serializing the numpy datatypes (`f8`, `b1`, `i8`, `O8`, `f4`) and then using those defined aliases.
```
base_Blendedness_abs_instFlux:
  dtype: &id001 !!python/object/apply:numpy.dtype
    args: [f8, 0, 1]
    state: !!python/tuple [3, <, null, null, null, -1, -1, 0]
base_ClassificationExtendedness_value:
  dtype: *id001
base_LocalBackground_flag:
  dtype: &id002 !!python/object/apply:numpy.dtype
    args: [b1, 0, 1]
    state: !!python/tuple [3, '|', null, null, null, -1, -1, 0]
base_LocalBackground_instFlux:
  dtype: *id001
base_LocalBackground_instFluxErr:
  dtype: *id001
base_PixelFlags_flag_bad:
  default: true
  dtype: *id002
base_PixelFlags_flag_crCenter:
  dtype: *id002
[...]
```

**Comment by wmwv** [comment]
@yymao Do we even need schema files for these readers?  Perhaps one resolution to this present issue is to remove the schema files and the parts of the reader that refer to them.

Every Parquet file within a DC2DMCatalog derived class has the same schema and the gcr class no longer has to fill in missing columns.

When I convert/re-generate the Object tables in future work, generating a consistent schema and storing the default values in the Parquet files will be part of that effort (under `DC2-production`).

**Comment by yymao** [comment]
@wmwv I was thinking exactly the same thing when I was modifying the reader, but I wasn't sure if every Parquet file has the same schema (but now you've confirmed it).

**Comment by wmwv** [comment]
I've approved the PR because it is the fastest way to fixing the problem and unblocking #301 and #296.

**Comment by yymao** [comment]
OK, I am going to merge #303. 

I propose that we remove these schema functionality when we make the transition to move the object catalogs to parquet (at which time we will also merge those readers).

**Comment by wmwv** [comment]
#304 created
